include/rocksdb/user_defined_index.h

```diff
@@ -163,6 +167,20 @@ class UserDefinedIndexFactory : public Customizable {
   // block
   virtual std::unique_ptr<UserDefinedIndexReader> NewReader(
       Slice& index_block) const = 0;
+
+  // New API for allowing customized comparator
+  virtual Status NewBuilder(std::unique_ptr<UserDefinedIndexBuilder>& builder,
+                            UserDefinedIndexOption& /*option*/) const {
```

Comment:
Input only arguments come first - C++ style guideline. Also we can make it const.